### PR TITLE
Fix bad test for unknown hook type

### PR DESCRIPTION
**Pull request by dstandish**
should hopefully fix broken main

### CODE REVIEW

**Comment by dstandish** [comment]
@Taragolis

**Comment by Taragolis** [comment]
You a bit faster create PR 😺 
It is a bit strange that it wouldn't failed earlier in https://github.com/apache/airflow/pull/25980

**Comment by dstandish** [comment]
haha 😝

File: tests/providers/common/sql/operators/test_sql.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def test_get_hook(self, mock_get_conn, database):
 
     def test_not_allowed_conn_type(self, mock_get_conn):
         mock_get_conn.return_value = Connection(conn_id='sql_default', conn_type='s3')
-        with pytest.raises(AirflowException, match=r"You are trying to use `common-sql`"):
+        with pytest.raises(AirflowException, match=r'Unknown hook type "s3"'):
```
Comment (Taragolis) [inline tests/providers/common/sql/operators/test_sql.py]: I thought better replace conn_type from `s3`, which not exists anymore to existed conn_type like `aws`, `google_cloud_platform`, `http` so we won't change actual test - provide use unsupported Hook

Comment (dstandish) [reply tests/providers/common/sql/operators/test_sql.py]: ohhhh... goooood point lemme see

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
> It is a bit strange that it wouldn't failed earlier in #25980

It has not failed because we have selective checks that only run porvider's tests for those "changed providers" and those that "depend" on them. 

Technically speaking we would run automatically "amazon" tests if "|common.sql" changed (because amazon depends on common.sql):

https://github.com/apache/airflow/blob/main/generated/provider_dependencies.json#L35

But we would not run "common.sql" tests when "amazon" changes because "common.sql" does not depend on amazon:

https://github.com/apache/airflow/blob/main/generated/provider_dependencies.json#L199

The problem in this case was that we had implicit dependency of common.sql "tests" on "amazon" provider - because in the common.sql we were using s3 as example connection, so there was a hidden dependency of "common.sql" tests on "amazon" :).

So actually failing main was our "canary" tests failing showing that there was a test that slipped through the safety net. This is the price to pay for the optimisations we have  (we run a small set of provider test now when only providers change  - that saves a lot of time and build hours).

The change now changed it to "fs", so we are good as "fs" is not a separate provider but it is part of Airlfow so the impiicit dependency is gone.

**Comment by Taragolis** [comment]
@potiuk Oh, very appreciate for detail explanation

image

**Pull request merged**